File: simulators/de.unistuttgart.iste.sqa.mpw.karasimulator.java/examples/src/main/java/de/unistuttgart/kara/examples/Example03.java
package de.unistuttgart.kara.examples;

import de.unistuttgart.kara.main.SimpleKaraGame;

public class Example03 extends SimpleKaraGame {
    public static void main(final String[] args) {
        createInstance(Example03.class);
    }

    /**
     * Another kara program, which tests the loading of another world.
     */
    @Override
    protected void run() {
        displayInNewGameWindow();
        loadWorldFromResourceFile("de.unistuttgart.kara.worlds/example03.ter");
        startGame();

        kara.turnRight();
        kara.move();
    }

}

File: examples/Example03.cpp
#include "SimpleKaraGame.h"

using namespace kara;

class Example03 : public karasimulator::SimpleKaraGame {
protected:
    void run() override;
};

void Example03::run() {
    loadWorldFromResourceFile("resources/worlds/example03.ter");
    startGame();

    kara.turnRight();
    kara.move();
}

int main() {
    karasimulator::SimpleKaraGame::createInstance<Example03>();
    return 0;
}
